### PR TITLE
[GLUTEN-1796][CH] Collect performance metrics from CH backend

### LINKED ISSUE #1796: [CH] Collect performance metrics from CH backend
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
Collect performance metrics from CH backend

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by zzcclp**
## What changes were proposed in this pull request?

Collect performance metrics from CH backend

Co-authored-by: liuneng1994 <neng.liu@kyligence.io>

(Fixes: #1796 )
Close #1796 

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: gluten-core/src/main/scala/io/glutenproject/substrait/SubstraitContext.scala [hunk 1]
```diff
@@ -72,6 +72,9 @@ class SubstraitContext extends Serializable {
   private val operatorToRelsMap = new java.util.HashMap[
     java.lang.Long, java.util.ArrayList[java.lang.Long]]()
 
+  // Only for debug conveniently
+  val operatorToPlanName = new java.util.HashMap[java.lang.Long, java.lang.String]()
```
Comment (baibaichen) [inline gluten-core/src/main/scala/io/glutenproject/substrait/SubstraitContext.scala:76]: Is it relly  need in the final code？

Comment (zzcclp) [reply gluten-core/src/main/scala/io/glutenproject/substrait/SubstraitContext.scala:76]: will move `PlanNodesUtil.scala` to ch backend in another pr, which is used only in ch backend.

Comment (zzcclp) [reply gluten-core/src/main/scala/io/glutenproject/substrait/SubstraitContext.scala:76]: will move `PlanNodesUtil.scala` into ch backend, which is used only in ch backend.

File: gluten-core/src/main/scala/io/glutenproject/utils/PlanNodesUtil.scala [hunk 2]
```diff
@@ -38,7 +38,7 @@ object PlanNodesUtil {
 
     // input
     val iteratorIndex: Long = context.nextIteratorIndex
-    var operatorId = context.nextOperatorId
+    var operatorId = context.nextOperatorId("ClickHouseBuildSideRelationReadIter")
```
[the remark this answers is not in the document]
Comment (zzcclp) [reply gluten-core/src/main/scala/io/glutenproject/utils/PlanNodesUtil.scala:41]: will move `PlanNodesUtil.scala` into ch backend, which is used only in ch backend.

File: gluten-core/src/main/scala/io/glutenproject/substrait/SubstraitContext.scala (hunk 1 above)
Comment (zzcclp) [reply gluten-core/src/main/scala/io/glutenproject/substrait/SubstraitContext.scala:76]: keep it in the code, will be used for debug later

**Review by rui-mo — DISMISSED** [review]

File: gluten-core/src/main/scala/io/glutenproject/substrait/SubstraitContext.scala (hunk 1 above)
Comment (rui-mo) [reply gluten-core/src/main/scala/io/glutenproject/substrait/SubstraitContext.scala]: OK, that looks good to me.

**Review by rui-mo — APPROVED** [review]

**Comment by baibaichen** [comment]
LGTM

**Pull request merged**